src/main.c

```diff
@@ -388,61 +88,7 @@ void main(void)
 		printk("Modem info could not be initialised: %d\n", err);
 		return;
 	}
+	nat_test_init();
 
-	err = setup_connection(&client_fd);
-	if (err < 0) {
-		printk("Failed to configure connection: %d\n", errno);
-		return;
-	}
-
-	while (current_protocol < NONE) {
-		err = send_data(client_fd, timeout_s);
-		if (err < 0) {
-			goto reconnect;
-		}
-
-		printk("Waiting for response...\n");
-
-		err = poll_and_read(client_fd, timeout_s);
-		if (err < 0) {
-			goto reconnect;
-		} else if (err == 0) {
-			timeout_s = get_timeout(true);
-		} else if (err > 0) {
-			timeout_s = get_timeout(false);
-		}
-
-		if (timeout_s == 0) {
-			current_protocol++;
-			if (strcmp(get_protocol_name(current_protocol),
-				   "NONE")) {
-				timeout_s = get_timeout(false);
-				goto reconnect;
-			} else {
-				err = close(client_fd);
-				if (err < 0) {
-				}
-				break;
-			}
-		}
-
-		continue;
-
-	reconnect:
-
-		err = close(client_fd);
-		if (err < 0) {
-		}
-
-		err = setup_connection(&client_fd);
-		if (err < 0) {
-			printk("Failed to reconfigure connection: %d\n", errno);
-			return;
-		}
-	}
-
-	printk("Finished NAT timeout measurements.\nUDP keep-alive time: %d seconds\nTCP keep-alive time: %d seconds\n",
-	       keep_alive_time.udp, keep_alive_time.tcp);
-
-	(void)close(client_fd);
+	nat_test_start(TEST_UDP_AND_TCP);
```

Comment:
Should this have error checking? If not, the function can  probably have `void` return type

Reply:
It doesn't really matter if I get an error here or not since the test will print to the same output regarding starting/error.
The return value becomes important when starting from shell.
I did add the error check for clarity though.